developer_manual/app/extstorage.rst

```diff
@@ -1,67 +1,86 @@
 =========================
-External storage backends
+External Storage Backends
 =========================
 
 .. sectionauthor:: Vincent Petry <pvince81@owncloud.com>
 
-Have a look at the source code of the `FTP external storage app
-<https://github.com/owncloud/files_external_ftp>`_ for a good example.
+This section shows how a standard app can provide external storage backends. 
 
-The following sections will indicate what modifications need to be done in
-a standard app to make it a provider of external storage backends.
+To do so, requires several steps. These are:
 
-Filesystem type
-===============
+- `Configure the filesystem type`_
+- `Implement the storage class(es)`_
+- `Create the backend adapter`_
+- `Register the backend adapter`_
+- `Test the storage backend`_
 
-The :file:`/appinfo/info.xml` must be adjusted to specify **filesystem** as type:
+.. NOTE:: 
+   To save time, however, you can learn from an existing example, by reading
+   through the source code of the `FTP external storage app
+   <https://github.com/owncloud/files_external_ftp>`_.
+
+Configure the filesystem type
+=============================
+
+First, the :file:`/appinfo/info.xml` must be adjusted to specify the ``type`` as:
+``filesystem``. For example:
 
 .. code-block:: xml
 
 .. include:: examples/storage-backend/appinfo/info.xml
 
-Implementing the storage
-========================
+Implement the storage class(es)
+===============================
 
-Usually one should implement the interface **\\OCP\\Files\\Storage\\IStorage**
-but the easiest way is to directly extend
-**\\OCP\\Files\\Storage\\StorageAdapter** which already provides the
-implementation for many commonly required methods.
+Next, you need to create a storage class. Usually, you should implement the
+interface ``\\OCP\\Files\\Storage\\IStorage``. But, the easiest way is to
+directly extend ``\\OCP\\Files\\Storage\\StorageAdapter``, as it already
+provides an implementation for many of the commonly required methods.
 
-For this storage, we'll use a fake library which we'll call **FakeStorageLib**.
-We create a class that implements all filesystem operations required by
-ownCloud.
+Here’s an example of how you would create one that implements all the
+filesystem operations required by ownCloud, using a fictitious library called
+``FakeStorageLib``.
 
 .. code-block:: php
 
 .. include:: examples/storage-backend/OCA/MyStorageApp/Storage/MyStorage.php
 
-For this example we simply mapped the storage methods to the one from the
-library. In many cases, the underlying library might not support some
+For this example we mapped the available storage methods to the ones from the
+library. Note that, in many cases, the underlying library might not support some
 operations and might need extra code to work this around.
 
-Stat cache
-----------
+.. NOTE::
```

Comment:
The following methods need to be implemented and are not provided by the storage adapter:
- file_exists
- filetype
- fopen
- getId
- mkdir
- opendir
- rmdir
- stat
- touch
- unlink

=> "it is good practice to implement all methods, however only the following are really required for the storage backend to work. But, it will likely not perform as well as it could."